index.ts

```diff
@@ -197,74 +197,27 @@ const _generateBrazeRequestBody = (pluginEvent: PluginEvent, meta: BrazeMeta): B
     }
 }
 
-export const exportEvents = async (pluginEvents: PluginEvent[], meta: BrazeMeta): Promise<void> => {
-    if (!pluginEvents.length) {
-        console.warn('Received `exportEvents` with no events.')
-        return
-    }
-
+export const onEvent = async (pluginEvent: PluginEvent, meta: BrazeMeta): Promise<void> => {
     // NOTE: We compute a unique ID for this request so we can identify the same request in the logs
-    const requestId = crypto.createHash('sha256').update(JSON.stringify(pluginEvents)).digest('hex')
+    const requestId = crypto.createHash('sha256').update(JSON.stringify(pluginEvent)).digest('hex')
     const startTime = Date.now()
-    let oldestEventTimestamp = Date.now()
-
-    const brazeRequestBodies = pluginEvents.map((pluginEvent) => {
-        if (pluginEvent.timestamp && new Date(pluginEvent.timestamp).getTime() < oldestEventTimestamp) {
-            oldestEventTimestamp = new Date(pluginEvent.timestamp).getTime()
-        }
-        return _generateBrazeRequestBody(pluginEvent, meta)
-    })
 
-    console.log(
-        `Braze plugin export, received ${pluginEvents.length} events. Exporting ${
-            brazeRequestBodies.length
-        } batches. Oldest event in this batch was received ${(Date.now() - oldestEventTimestamp) / 1000} seconds ago.`,
-        requestId
-    )
-
-    if (
-        brazeRequestBodies.length === 0 ||
-        brazeRequestBodies.every((body) => body.attributes.length === 0 && body.events.length === 0)
-    ) {
-        return console.log('No events to export.')
+    const brazeRequestBody = _generateBrazeRequestBody(pluginEvent, meta)
+    if (brazeRequestBody.attributes.length === 0 && brazeRequestBody.events.length === 0) {
+        return
```

Comment:
@tiina303 @paolodamico Maybe it's a good idea to log here something:`Exit on the empty event: ${eventName}`

Reply:
that could be quite noisy, no?

Reply:
> that could be quite noisy, no?
Not sure, to be honest. How often it can be that a request contains no events and no attributes?

Reply:
see https://posthog.slack.com/archives/C0386ASS3TN/p1695123530196279?thread_ts=1692342517.071949&cid=C0386ASS3TN for more discussion for your team's specifics